### ISSUE TITLE
Example Karpenter does not work, karpenter pods are pending (version 20.4.0)

### ISSUE THREAD

**Issue by kartoch** [terraform-aws-modules/terraform-aws-eks#2943]
## Description

When the karpenter example is applied, karpenter pods are pending because of taints.
 
When changed the karpenter helm values, it works:
```
resource "helm_release" "karpenter" {

  # values ommited

  values = [
    <<-EOT
    # values ommited
    tolerations:
      - effect: NoSchedule
        operator: "Equal"
        key: eks.amazonaws.com/compute-type
        value: fargate
    EOT
  ]
}
```

- [X] ✋ I have searched the open/closed issues and my issue is not listed.

## Versions

- Module version: 20.4.0

- Terraform version: 
```
Terraform v1.6.6
```
- Provider version(s):
```
Terraform v1.6.6
on linux_amd64
+ provider registry.terraform.io/gavinbunney/kubectl v1.14.0
+ provider registry.terraform.io/hashicorp/aws v5.38.0
+ provider registry.terraform.io/hashicorp/cloudinit v2.3.3
+ provider registry.terraform.io/hashicorp/helm v2.12.1
+ provider registry.terraform.io/hashicorp/time v0.10.0
+ provider registry.terraform.io/hashicorp/tls v4.0.5
```

## Reproduction Code [Required]

```
> cd examples/karpenter/
> terraform apply -approve
# get kubectl credentials
> k get pods - n karpenter
karpenter     karpenter-b8796757b-7gpr2   0/1     Pending   0   17m
karpenter     karpenter-b8796757b-nvw6k   0/1    Pending  0   17m
```

## Expected behavior

karpenter pods running.

## Actual behavior

karpenter pods in pending.

**Comment by auyer** [comment]
I am having a similar error. When I check the Events in the Karpenter pod, this is what I get:
```
Warning  FailedScheduling  15m   fargate-scheduler  Misconfigured Fargate Profile: fargate profile platform-k8s-sandbox-fargate blocked for new launches due to: Pod execution role is not found in auth config or does not have all re │
│ quired permissions for launching fargate pods.
```

Is this the same as you, @kartoch ?

**Comment by auyer** [comment]
Could this be related to #2912 ?

**Comment by rkapil** [comment]
> ## Description
> When the karpenter example is applied, karpenter pods are pending because of taints.
> 
> When changed the karpenter helm values, it works:
> 
> ```
> resource "helm_release" "karpenter" {
> 
>   # values ommited
> 
>   values = [
>     <<-EOT
>     # values ommited
>     tolerations:
>       - effect: NoSchedule
>         operator: "Equal"
>         key: eks.amazonaws.com/compute-type
>         value: fargate
>     EOT
>   ]
> }
> ```
> 
> * [x]  ✋ I have searched the open/closed issues and my issue is not listed.
> 
> ## Versions
> * Module version: 20.4.0
> * Terraform version:
> 
> ```
> Terraform v1.6.6
> ```
> 
> * Provider version(s):
> 
> ```
> Terraform v1.6.6
> on linux_amd64
> + provider registry.terraform.io/gavinbunney/kubectl v1.14.0
> + provider registry.terraform.io/hashicorp/aws v5.38.0
> + provider registry.terraform.io/hashicorp/cloudinit v2.3.3
> + provider registry.terraform.io/hashicorp/helm v2.12.1
> + provider registry.terraform.io/hashicorp/time v0.10.0
> + provider registry.terraform.io/hashicorp/tls v4.0.5
> ```
> 
> ## Reproduction Code [Required]
> ```
> > cd examples/karpenter/
> > terraform apply -approve
> # get kubectl credentials
> > k get pods - n karpenter
> karpenter     karpenter-b8796757b-7gpr2   0/1     Pending   0   17m
> karpenter     karpenter-b8796757b-nvw6k   0/1    Pending  0   17m
> ```
> 
> ## Expected behavior
> karpenter pods running.
> 
> ## Actual behavior
> karpenter pods in pending.

I am seeing the same issue, karpenter pods in pending. Just pulled from master today.

**Comment by invalidred** [comment]
Also seeing the same issue, Karpenter pods are in pending state with error

>  Misconfigured Fargate Profile: fargate profile karpenter blocked for new launches due to: Pod execution role is not found in auth config or does not have all required permissions for launching fargate pods

**Comment by antonbabenko** [comment]
This issue has been resolved in version 20.8.3 :tada:
